src/Chapter2.hs

```diff
@@ -75,791 +77,774 @@ When working with lists, the most practical module will be "Data.List":
+-- provides the "String" alias to '[Char]'. So, in some places, you will
+-- see 'String', and in others, you will see '[Char]', but they mean the
+-- same thing. We will explain better how it works in Chapter Three. For
+-- now, you only need to know that you can use 'String' and '[Char]'
+-- interchangeably.
+--
+-- To create a list, you need to put elements in square brackets (the
+-- same as used for the list type) and separate them by commas. Such
+-- expressions are called __list literals__. For example, the expression
+-- "[3, 5, 1]" creates a list of three numbers, where the first list
+-- element is number 3, and the last element is number 1. Similarly, you
+-- can create a list of two booleans: [False, True]. A list without
+-- elements is just [].
+--
+-- You probably noticed that lists could be of any type of
+-- elements. Often you want to write a function that works with lists of
+-- any type (but consistent inside one list). This feature is called
+-- __parametric polymorphism__. It will be explained in more detail
+-- later, but when working with lists, you often will see type signatures
+-- like:
+--
+-- @
+-- foo :: [a] -> [b] -> [a]
+-- @
+--
+-- The above type signature means that this function takes two lists: one
+-- with elements of some type "a" and another with elements of type "b"
+-- (the function doesn't care about the specific types) and returns the
+-- list with elements of the same type as the first list. Such words "a"
+-- and "b" are called __type variables__.
+--
+-- For comparison, specific types in Haskell start with an uppercase
+-- letter (Int, Bool, Char, etc.), where type variables begin with a
+-- lowercase letter (a, b, el, etc.). This is the way to distinguish
+-- between these types.
+--
+-- The Haskell standard library already provides a lot of functions to
+-- work with lists. And you will need to operate a lot with standard
+-- functions in the upcoming exercises. Remember, Hoogle is your friend!
+
+-- |
+-- =⚔️= Task 1
+--
+-- Explore lists by checking types of various list expressions and
+-- functions in GHCi and insert the corresponding resulting output below:
+--
+-- List of booleans:
+-- >>> :t [True, False]
+-- [True, False] :: [Bool]
+--
+-- String is a list of characters:
+-- >>> :t "some string"
+-- "some string" :: String
+--
+-- Empty list:
+-- >>> :t []
+-- [] :: [a]
+--
+-- Append two lists:
+-- >>> :t (++)
+-- (++) :: [a] -> [a] -> [a]
+--
+-- Prepend an element at the beginning of a list:
+-- >>> :t (:)
+-- (:) :: a -> [a] -> [a]
+--
+-- Reverse a list:
+-- >>> :t reverse
+-- reverse :: [a] -> [a]
+--
+-- Take first N elements of a list:
+-- >>> :t take
+-- take :: Int -> [a] -> [a]
+--
+-- Create a list from N same elements:
+-- >>> :t replicate
+-- replicate :: Int -> a -> [a]
+--
+-- Split a string by line breaks:
+-- >>> :t lines
+-- lines :: String -> [String]
+--
+-- Join a list of strings with line breaks:
+-- >>> :t unlines
+-- unlines :: [String] -> String
+-- =⚔️= Task 2
```

Comment:
I think tests fail because this line should be separated from the doctests results.
Generally, I see that you reformatted all the comments, so it is a bit harder to check the changes that you made in the file, I usually recommend to just keep relevant code changes, so it is faster for both of us to communicate, but it is not a big deal 🙂 👍🏼